### PR TITLE
Implement spawn using wasix crate

### PR DESCRIPTION
**Pull request by g-bartoszek**
Implements `Command::spawn` using wasix `proc_spawn`

### CODE REVIEW

File: library/std/src/sys/wasix/process/process_wasix.rs [hunk 1]
```diff
@@ -2,15 +2,33 @@
...
+        wasi::OPTION_SOME => Some(unsafe { AnonPipe::from_raw_fd(option_fd.u.some as i32) }),
+        _ => None,
+    }
+}
+
+fn to_wasi_mode(input: &Stdio) -> wasi::StdioMode {
+    match input {
+        Stdio::Inherit => wasi::STDIO_MODE_INHERIT,
+        Stdio::MakePipe => wasi::STDIO_MODE_PIPED,
+        _ => wasi::STDIO_MODE_NULL,
```
Comment (g-bartoszek) [inline library/std/src/sys/wasix/process/process_wasix.rs:29]: Should we support other modes here?

File: library/std/src/sys/wasix/process/process_wasix.rs [hunk 2]
```diff
@@ -207,115 +159,173 @@ impl Process {
...
+                    Some(self.0.u.exit_signal.signal.raw() as i32)
+                }
+                _ => None,
+            }
+        }
     }
 
     pub fn core_dumped(&self) -> bool {
-        libc::WIFSIGNALED(self.0) && libc::WCOREDUMP(self.0)
+        self.code().map(|c| libc::WIFSIGNALED(c) && libc::WCOREDUMP(c)).unwrap_or(false)
```
Comment (g-bartoszek) [inline library/std/src/sys/wasix/process/process_wasix.rs:265]: I'm not sure if it makes sens to call `libc::WIFSIGNALED`/`libc::WCOREDUMP` on a value returned from `proc_spawn`

File: library/std/src/sys/wasix/process/process_wasix.rs (hunk 1 above)
Comment (theduke) [reply library/std/src/sys/wasix/process/process_wasix.rs:29]: These are actually all the available modes at the moment, so nothing to do on our side.

File: library/std/src/sys/wasix/process/process_wasix.rs (hunk 2 above)
Comment (theduke) [reply library/std/src/sys/wasix/process/process_wasix.rs:265]: Looks sensible to me.

**Comment by syrusakbary** [comment]
We will merge, although we need a bit more testing that we'll tackle in the next week.

Thanks a lot for the work on the PR @g-bartoszek

**Pull request merged**